frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -1163,4 +1154,88 @@ private String getProjectDefaultLicenseInfoHeaderText() {
+        for (int i = startIdx; i < count; i++) {
+            if (i < sortedProjects.size()) {
+                JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+                Project project = sortedProjects.get(i);
+                jsonObject.put("id", project.getId());
+                jsonObject.put("DT_RowId", project.getId());
+                jsonObject.put("name", SW360Utils.printName(project));
+                jsonObject.put("desc", project.isSetDescription() ? project.getDescription() : "");
+                jsonObject.put("state", project.isSetState() ? project.getState().toString() : "");
+                jsonObject.put("cState", project.isSetClearingState() ? project.getClearingState().toString() : "");
+                jsonObject.put("clearing", "Not loaded yet");
+                jsonObject.put("resp", project.isSetProjectResponsible() ? project.getProjectResponsible() : "");
+                jsonObject.put("lProjSize", String.valueOf(project.getLinkedProjectsSize()));
+                jsonObject.put("lRelsSize", String.valueOf(project.getReleaseIdToUsageSize()));
+                jsonObject.put("attsSize", String.valueOf(project.getAttachmentsSize()));
+                projectData.put(jsonObject);
+            }
+        }
+
+        return projectData;
+    }
+
+    private List<Project> sortProjectList(List<Project> projectList, Map<String, Integer> projectParameter) {
+        List<Project> sortedList;
+
+        switch (projectParameter.get(DATATABLE_SORT_COLUMN)) {
+            case 0: // Project Name
+                sortedList = projectList.stream().sorted(Comparator.comparing(
+                        p -> SW360Utils.printName(p).toLowerCase())).collect(Collectors.toList());
+                break;
+            case 1: // Description
+                sortedList = projectList.stream().sorted(Comparator.comparing(
+                        p -> p.isSetDescription() ? p.getDescription() : null,
+                        Comparator.nullsFirst(String::compareToIgnoreCase))).collect(Collectors.toList());
+                break;
+            case 2: // Project Responsible
+                sortedList = projectList.stream().sorted(Comparator.comparing(
+                        p -> p.isSetProjectResponsible() ? p.getProjectResponsible() : null,
+                        Comparator.nullsFirst(String::compareToIgnoreCase))).collect(Collectors.toList());
+                break;
+            case 3: // State
+                sortedList = projectList;
+                break;
+            case 4: // Clearing Status
+                sortedList = projectList;
+                break;
+            case 5: // Action
+                sortedList = projectList;
+                break;
+            default:
+                sortedList = projectList;
+        }
+
+        if (projectParameter.get(DATATABLE_SORT_DIRECTION) == 0) {
+            return sortedList;
+        } else {
+            return Lists.reverse(sortedList);
```

Comment:
For performance it might be better to sort by inverse order instead of reversing after sort.